core-rust/core-api-server/core-api-schema.yaml

```diff
@@ -926,12 +926,27 @@ components:
       minLength: 64
       maxLength: 64
       description: The hex-encoded ledger-wrapped transaction hash. This is known as the Ledger Hash. This hash is `SHA256(SHA256(ledger_transaction_bytes))`
+    ErrorResponseType:
+      type: string
+      enum:
+        - Basic
+        - TransactionSubmit
     ErrorResponse:
       type: object
+      discriminator:
```

Comment:
OK, there is some problem here. For reasons unknown to me yet C# OAS generator generates `public BasicErrorResponse(ErrorResponseType errorType = "BasicErrorResponse", int code = ...`, i.e. it tries to use string rather enum.

Reply:
OK, I think I know what the problem is. Current model tries to effectively serialize error responses to:
```
{"error_type":"Basic", "code":123, "message":"bleh"}
{"error_type":"TransactionSubmit", "code":321, "message":"bleh bleh", "some_extra_prop":321}
```
While this is clearly better and more logical structure I'd suggest to go back to what we have with Olympia, i.e.
```
{"code":123, "message":"bleh", "details":{"error_type":"Basic"}}
{"code":321, "message":"bleh bleh", "details": {"error_type":"TransactionSubmit", "some_extra_prop":321}}
```

At least this will work with OAS generator.

Reply:
Have agreed to proceed with the inheritance model - and will fix up the bugs in the C# open api generator on the C# end.